product-mini/platforms/nuttx/CMakeLists.txt

```diff
@@ -129,6 +129,10 @@ if(CONFIG_INTERPRETERS_WAMR_GLOBAL_HEAP_POOL)
   set(WAMR_BUILD_GLOBAL_HEAP_SIZE ${_HEAP_SIZE_})
 endif()
 
+if (CONFIG_INTERPRETERS_WAMR_MEM_ALLOC_WITH_USAGE)
+  set(WAMR_BUILD_MEM_ALLOC_WITH_USAGE 1)
```

Comment:
How about `CONFIG_INTERPRETERS_WAMR_ALLOC_WITH_USAGE` and `WAMR_BUILD_ALLOC_WITH_USAGE`?